### PR TITLE
Feat/scrapy

### PR DESCRIPTION
**Pull request by clairesquires**

### CODE REVIEW

File: src/app/utils/python/chunkText.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    end_pos = start_pos + window_size
+    chunks = []
+    while end_pos < len(string):
+        chunk = string[start_pos:end_pos]
+        chunks.append(chunk)
+        start_pos += window_jump
+        end_pos = start_pos + window_size
+    if end_pos > len(string):
+        end_chunk = string[start_pos:-1]
+        chunks.append(end_chunk)
```
Comment (matt-newhall) [inline src/app/utils/python/chunkText.py:16]: what happens in the case that `end_pos == len(string)` ?

File: src/app/utils/python/chunkText.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    # Returns:
+    #   jsons: The list of JSON strings with attributes url and input_text, where input_text is a chunk of the artcile_info_text with length window_size. (Note: return list of JSON strings rather than JSON objects)
+    jsons = []
+    window_size = 30
+    window_jump = 10
+    for page in scraped_pages:
+        chunks = chunkString(page['article_info_text'],
+                             window_size, window_jump)
+        for chunk in chunks:
+            d = {}
```
Comment (matt-newhall) [inline src/app/utils/python/chunkText.py:32]: better variable name than `d`?

File: src/app/utils/python/chunkText.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        {'url': "https://url_A",
+         'article_info_text': "As an AI language model, I'm not capable of personal experiences."},
+        {'url': "https://url_B",
+         'article_info_text': "However, I can tell you that my purpose is to assist and provide information."},
+        {'url': "https://url_C",
+         'article_info_text': "I am constantly learning and improving my responses."}
+    ]
+    jsons = chunkText(test_jsons)
+    # for j in jsons:
+    #     print(j)
```
Comment (matt-newhall) [inline src/app/utils/python/chunkText.py:52]: can be removed now! :)

File: src/app/utils/python/chunkText.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        chunks.append(chunk)
+        start_pos += window_jump
+        end_pos = start_pos + window_size
+    if end_pos > len(string):
+        end_chunk = string[start_pos:-1]
+        chunks.append(end_chunk)
+    return chunks
+
+
+def chunkText(scraped_pages: List[Dict[str, str]]) -> List[str]:
```
Comment (matt-newhall) [inline src/app/utils/python/chunkText.py:20]: Does `Dict[str, str]` state that the key is a `str` and the value is also `str` or that the dict is of size 2?

File: src/app/utils/python/scraper.py [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    def process_item(self, item):
+        self.scraped_items.append(item)
+        return item
+
+    def spawn(self, **kwargs):
+        self.process.crawl(
+            MySpider, output_callback=self.process_item, **kwargs)
+
+    def run(self):
+        self.process.start()
```
Comment (matt-newhall) [inline src/app/utils/python/scraper.py:39]: are we strict typing all functions?

File: src/app/utils/python/chunkText.py (hunk 4 above)
Comment (clairesquires) [reply src/app/utils/python/chunkText.py:20]: This means that the key and value are both of type str

File: src/app/utils/python/chunkText.py (hunk 3 above)
Comment (clairesquires) [reply src/app/utils/python/chunkText.py:52]: Thanks!

**Review by matt-newhall — APPROVED** [review]
lavely :))

**Pull request merged**